app/views/pieces/show.html.erb

```diff
@@ -22,9 +22,14 @@
 
   <% if current_user && current_user.id == @user.id %>
     <div>
+      <%= link_to 'addToCollection', add_to_collection_user_collection_piece_path(@user.id, @collection.id, @piece.id), id: "addToCollection" %> |
```

Comment:
Will have to remove this from the conditional here to allow all users to be able to add to their collections.